src/samples/network/network-status.c

```diff
@@ -92,6 +93,8 @@ _on_network_event(void *data, const struct sol_network_link *link, enum sol_netw
 {
     char *name;
 
+    SOL_NETWORK_LINK_CHECK_VERSION(link);
```

Comment:
This is a callback that gives you the type, no need to check it. 

Same for others in this file